### PR TITLE
adds support for service status

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for service status

## Why are we making these changes?

We would like to explicit report the status of a service.

service-status

### CODE REVIEW

File: waiter/src/waiter/service.clj [hunk 1]
```diff
@@ -293,3 +293,19 @@
...
                  ^Runnable (fn [] (cid/with-correlation-id correlation-id (start-fn)))))
       (log/info service-id "has been started on another thread" cache-value))))
+
+(defn resolve-service-status
+  "Determines the service status at any point in time.
+   A service can be one of the following states:
+   - Starting: the service has no healthy instances and is starting one up,
+   - Running: the service is successfully running at its desired scale,
+   - Failing: the service has instances failing to start,
+   - Waiting: the service is waiting to be started,
```
Comment (dposada) [inline waiter/src/waiter/service.clj]: How about "Idle" instead of "Waiting"?

File: waiter/src/waiter/service.clj [hunk 2]
```diff
@@ -293,3 +293,19 @@
...
+   - Scaling: the service is being scaled."
```
Comment (DaoWen) [inline waiter/src/waiter/service.clj]: Having both "Running" and "Scaling" seems confusing. We probably shouldn't differentiate between these two states in the Status.

File: waiter/src/waiter/service.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/service.clj]: Done.

File: waiter/src/waiter/service.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/service.clj]: Used a single "Running" status.

**Pull request merged**